File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    p_arr = deque(priorities)
    name_arr = deque(range(len(priorities)))
    
    while True:
        x = p_arr.popleft()
        idx = name_arr.popleft()
        if p_arr and x < max(p_arr):
            p_arr.append(x)
            name_arr.append(idx)
        else:
            answer += 1
            if idx == location:
                break
    
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        Deque<int[]> q = new ArrayDeque<>();
        
        for(int i = 0; i<priorities.length; i++){
            q.add(new int[]{i, priorities[i]});
        }
        
        int cnt = 0;
        while(!q.isEmpty()){
            int[] cur = q.poll();
            boolean isTrue = false;
            
            for(int i = 0; i<q.size(); i++){
                int[] next = q.poll();
                if(cur[1] < next[1]){
                    isTrue = true;
                }
                q.add(next);
            }
            
            if(!isTrue){
                cnt++;
                if(cur[0] == location){
                    break;
                }
            }else{
                q.add(cur);
            }
        }
        
        return cnt;
        
    }
}